prompt_poet/prompt.py

```diff
@@ -451,6 +454,13 @@ def _render_parts(self):
                     "Tokens were provided in template. Regular tokenization will be skipped."
                 )
                 self._total_tokens += len(part.tokens)
+    
+    def _validate_template_replacements(self, yaml_part: PromptPart):
+        """Validate that all required keys in the expected_template_data_keys are present in template_data."""
+        if yaml_part.expected_template_data_keys:
+            missing_keys = [key for key in yaml_part.expected_template_data_keys if key not in self._template_data]
+            if missing_keys:
+                raise ValueError(f"Missing replacement values for keys: {missing_keys}")
```

Comment:
nit

raise ValueError(f"Missing replacement values for keys: {missing_keys=}")